server/db/db.js

```diff
@@ -1,826 +1,882 @@
-            }
-        }
-    }
-
-    return null
-}
-
-module.exports.saveHeartbeat = async function(systemId, flicLastSeenTime, flicLastPingTime, heartbeatLastSeenTime, client) {
-    const transactionMode = (typeof client !== 'undefined')
+  try {
+    if (!transactionMode) {
+      client = await pool.connect()
+    }
 
-    try {
-        if(!transactionMode) {
-            client = await pool.connect()
-        }
-        
-        let { rows } = await client.query("SELECT * FROM hubs WHERE system_id = $1 LIMIT 1", [systemId])
-        if(rows.length === 0) {
-            throw new Error("Tried to save a heartbeat for a hub that doesn't exist yet.")
-        }
-
-        const query = "UPDATE hubs SET flic_last_seen_time = $1, flic_last_ping_time = $2, heartbeat_last_seen_time = $3 WHERE system_id = $4"
-        const values = [flicLastSeenTime, flicLastPingTime, heartbeatLastSeenTime, systemId]
-        await client.query(query, values) 
-    }
-    catch(e) {
-        helpers.log(`Error running the saveHeartbeat query: ${e}`);
-    } finally {
-        if (!transactionMode) {
-            try {
-                client.release()
-            } catch(err) {
-                helpers.log(`saveHeartbeat: Error releasing client: ${err}`)
-            }
-        }
-    }
-}
-
-module.exports.saveHubRename = async function(systemId, systemName, client) {
-    const transactionMode = (typeof client !== 'undefined')
+    const { rows } = await client.query('SELECT * FROM hubs WHERE system_id = $1 LIMIT 1', [systemId])
+    if (rows.length === 0) {
+      throw new Error("Tried to save a heartbeat for a hub that doesn't exist yet.")
+    }
 
-    try {
-        if(!transactionMode) {
-            client = await pool.connect()
-        }
-        
-        let { rows } = await client.query("SELECT * FROM hubs WHERE system_id = $1 LIMIT 1", [systemId])
-        if(rows.length === 0) {
-            throw new Error("Tried to rename a hub that doesn't exist yet.")
-        }
-
-        const query = "UPDATE hubs SET system_name = $1 WHERE system_id = $2"
-        const values = [systemName, systemId]
-        await client.query(query, values) 
-    }
-    catch(e) {
-        helpers.log(`Error running the saveHubRename query: ${e}`);
-    } finally {
-        if (!transactionMode) {
-            try {
-                client.release()
-            } catch(err) {
-                helpers.log(`saveHubRename: Error releasing client: ${err}`)
-            }
-        }
-    }
-}
-
-module.exports.saveHubMuteStatus = async function(systemId, muted, client) {
-    const transactionMode = (typeof client !== 'undefined')
+    const values = [flicLastSeenTime, flicLastPingTime, heartbeatLastSeenTime, systemId]
+    await client.query(
+      'UPDATE hubs SET flic_last_seen_time = $1, flic_last_ping_time = $2, heartbeat_last_seen_time = $3 WHERE system_id = $4',
+      values,
+    )
```

Comment:
Maybe in a future refactor we can move the values array into the function call and save a line?